File: Matlab/MovingAvg.m
function m_series=movingAvg(s,n)
    if(nargin<2)
       n=5; 
    end
    
    temp_series=0;
     m_series=zeros(size(s,1),size(s,2)-(n-1));

    for index=1:size(s,1)
        for i=1:size(s,2)-n+1
            temp_series=[temp_series mean(s(index,i:i+(n-1)))];
        end
        m_series(index,:)=temp_series;

        temp_series=0;
    end

end

File: Python/lib/MovingAvg.py
# -*- coding: utf-8 -*-
import numpy as np

# Input is a same length series
def MovingAvg(series, n=5):
    # Check for series size. (Example: array([1,2,3])
    if len(series.shape) == 1:
        series = series.reshape(1, series.shape[0])
    result = np.zeros((series.shape[0], series.shape[1]- n + 1))

    for idx in range(series.shape[1] - n + 1):
        result[:, idx] = np.mean(series[:, idx:idx+n], 1)

    return result

# Testing Code
if __name__ == '__main__':
    MovingAvg(np.array([[1,2,3,4,5,4,3,2,1], [5,4,3,2,1,2,3,4,5]]), 3)
